extensions/sql/contract-definition/loader/src/main/java/org/eclipse/dataspaceconnector/sql/contractdefinition/loader/SqlContractDefinitionLoader.java

```diff
@@ -0,0 +1,118 @@
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Daimler TSS GmbH - Initial API and Implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.sql.contractdefinition.loader;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.eclipse.dataspaceconnector.dataloading.ContractDefinitionLoader;
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.persistence.EdcPersistenceException;
+import org.eclipse.dataspaceconnector.spi.transaction.TransactionContext;
+import org.eclipse.dataspaceconnector.spi.types.domain.contract.offer.ContractDefinition;
+import org.eclipse.dataspaceconnector.sql.contractdefinition.schema.SqlContractDefinitionTables;
+
+import java.sql.Connection;
+import java.util.Objects;
+import javax.sql.DataSource;
+
+import static org.eclipse.dataspaceconnector.sql.SqlQueryExecutor.executeQuery;
+
+public class SqlContractDefinitionLoader implements ContractDefinitionLoader {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private final DataSource dataSource;
+    private final TransactionContext transactionContext;
+
+    public SqlContractDefinitionLoader(DataSource dataSource, TransactionContext transactionContext) {
+        this.dataSource = Objects.requireNonNull(dataSource);
+        this.transactionContext = Objects.requireNonNull(transactionContext);
+    }
+
+    @Override
+    public void accept(ContractDefinition definition) {
+        Objects.requireNonNull(definition);
+
+        String query = String.format("INSERT INTO %s (%s, %s, %s, %s) VALUES (?, ?, ?, ?)",
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_TABLE,
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_COLUMN_ID,
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_COLUMN_ACCESS_POLICY,
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_COLUMN_CONTRACT_POLICY,
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_COLUMN_SELECTOR);
+
+        transactionContext.execute(() -> {
+            try (Connection connection = dataSource.getConnection()) {
+                executeQuery(connection, query,
+                        definition.getId(),
+                        writeObject(definition.getAccessPolicy()),
+                        writeObject(definition.getContractPolicy()),
+                        writeObject(definition.getSelectorExpression()));
+            } catch (Exception exception) {
+                throw new EdcPersistenceException(exception);
+            }
+        });
+
+    }
+
+    private String writeObject(Object object) {
+        try {
+            String className = object.getClass().getName();
+            String content = OBJECT_MAPPER.writeValueAsString(object);
+
+            Envelope envelope = new Envelope(className, content);
+
+            return OBJECT_MAPPER.writeValueAsString(envelope);
```

Comment:
why are you wrapping JSON inside JSON? First, I don't understand the need for having the classnames in the database, and second, having JSON in a STRING field would probably complicate things like indexing and (full-text) searching. 
And even if you were hell-bent on having JSON in the database, why not just use a `Map<String, Object` with one entry, and serialize that?